developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,219 @@
+    uint32_t freq_96000  : 1;
+    uint32_t freq_176400 : 1;
+    uint32_t freq_192000 : 1;
+    uint32_t reserved_   : 16;
+  };
+
+  struct sample_sizes {
+    uint16_t bits_8    : 1;
+    uint16_t bits_16   : 1;
+    uint16_t bits_24   : 1;
+    uint16_t bits_32   : 1;
+    uint16_t bits_64   : 1;
+    uint16_t reserved_ : 11;
+  };
+
+  struct sample_containers {
+    uint16_t bits_8    : 1;
+    uint16_t bits_16   : 1;
+    uint16_t bits_32   : 1;
+    uint16_t bits_64   : 1;
+    uint16_t reserved_ : 12;
+  };
+
+  struct channel_configurations {
+    // FRONT_CENTER
+    uint32_t channel_mono : 1;
+
+    // FRONT_LEFT | BACK_LEFT
+    uint32_t channel_dual_mono : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT
+    uint32_t channel_stereo : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | LOW_FREQUENCY
+    uint32_t channel_2_1 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER
+    uint32_t channel_3_0 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | BACK_LEFT  | BACK_RIGHT
+    uint32_t channel_quad : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_CENTER
+    uint32_t channel_surround : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY
+    uint32_t channel_3_1 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT
+    uint32_t channel_5_0 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | SIDE_LEFT | SIDE_RIGHT
+    uint32_t channel_5_0_surround : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT
+    uint32_t channel_5_1 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | SIDE_LEFT | SIDE_RIGHT
+    uint32_t channel_5_1_surround : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+    uint32_t channel_7_0 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+    uint32_t channel_7_0_surround : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER
+    uint32_t channel_7_1 : 1;
+
+    // FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT
+    uint32_t channel_7_1_surround : 1;
+
+    uint32_t reserved_ : 16;
+  };
+
+| **mod_cfg**: It is generated by fw for each different stream format in real case, such as 48khz, 16bit, 4channel. Use should copy
+|              these config to module entry. If there is no mod_cfg built by fw, the mod_cfg can be skipped. The definition is:
+
+.. code-block:: bash
+
+   struct ModuleConfig {
+    uint32_t    par[4];
+    uint32_t    is_bytes;
+    uint32_t    cps;
+    uint32_t    ibs;
+    uint32_t    obs;
+    uint32_t    module_flags;
+    uint32_t    cpc;
+    uint32_t    obls;
+   };
+
+   par: Configuration lookup parameters used by the host driver
+   is_byte: Number of bytes required by the module instance
```

Comment:
is this RAM? Stack? Or text? Also is there a stack field?

Reply:
this is RAM which is for module instance initialization